suite/tests/CMakeLists.txt

```diff
@@ -3143,6 +3143,13 @@ if (UNIX)
 
     tobuild(linux.signest linux/signest.c)
     link_with_pthread(linux.signest)
+
+    if (UNIX)
+      tobuild_ops(linux.sigmangle_asynch linux/sigmangle_asynch.c "-vm_base 0x100000000 -no_vm_base_near_app" "")
```

Comment:
This asynch xl8 problem is not limited to signals nor to UNIX: our synchall, or an app suspending one of its threads, can hit this on Windows.  At least add a XXX comment about testing it on Windows too.  The idea about using a client w/ the synchall API would be a cross-platform test.

Also, I don't think "sig" is adding to the name: how about just "mangle_asynch" or "asynch_mangle".

Reply:
I'll change the test's name. I am currently working on a client test using synchall and will try if that exposes the bug.

Reply:
Done